packages/core/js-client/src/keypair/index.ts

```diff
@@ -33,7 +33,11 @@ export class KeyPair {
         return this.libp2pPeerId;
     }
 
-    constructor(private key: PrivateKey, private libp2pPeerId: PeerId) {}
+    constructor(
+        private privateKey: PrivateKey | undefined,
+        private publicKey: PublicKey,
```

Comment:
Why pass public key? It can be derived from private key. I'm not even sure why `libp2pPeerId` was passed, it also should be derived.

What would happen if you pass `privateKey` from one keypair, `publicKey` from a second keypair, and `peerId` from a third keypair?